src/common/commonutils/FileUtils.c

```diff
@@ -448,463 +563,172 @@ int SetDirectoryAccess(const char* directoryName, unsigned int desiredOwnerId, u
+                    skipLine = false;
                 }
-            }
 
-            if (false == matchFound)
-            {
-                OsConfigLogInfo(log, "SetFileSystemMountingOption: mount directory '%s' and/or mount type '%s' not found in either '%s' or '%s', nothing to remediate",
-                    mountDirectory ? mountDirectory : "-", mountType ? mountType : "-", fsMountTable, mountTable);
+                fclose(tempHandle);
             }
-        }
-        else
-        {
-            if (0 == (status = errno))
+            else
             {
-                status = ENOENT;
+                OsConfigLogError(log, "ReplaceMarkedLinesInFile: failed to create temporary file '%s'", tempFileName);
+                status = EACCES;
             }
-            OsConfigLogError(log, "SetFileSystemMountingOption: could not open '%s', setmntent() failed (%d)", fsMountTable, status);
-        }
 
-        if (0 == status)
+            fclose(fileHandle);
+        }
+        else
         {
-            // Copy from the manually built temp mount file one to the temp mount file two using the *mntent API to ensure correct format
-            if (0 == (status = CopyMountFile(tempFileNameOne, tempFileNameTwo, log)))
-            {
-                // When done assembling the final temp mount file two, move it in an atomic step to real mount file
-                rename(tempFileNameTwo, fsMountTable);
-            }
+            OsConfigLogError(log, "ReplaceMarkedLinesInFile: cannot read from '%s'", fileName);
+            status = EACCES;
         }
-
-        remove(tempFileNameOne);
-        remove(tempFileNameTwo);
+    }
+    else
+    {
+        OsConfigLogError(log, "ReplaceMarkedLinesInFile: out of memory");
+        status = ENOMEM;
     }
 
-    FREE_MEMORY(newLine);
-    FREE_MEMORY(tempFileNameOne);
-    FREE_MEMORY(tempFileNameTwo);
-
-    return status;
-}
-
-static unsigned int GetNumberOfCharacterInstancesInFile(const char* fileName, char what)
-{
-    unsigned int numberOf = 0;
-    FILE* file = NULL;
-    int fileSize = 0;
-    int i = 0;
-    int next = 0;
+    FREE_MEMORY(line);
 
-    if (FileExists(fileName))
+    if (0 == status)
     {
-        if (NULL != (file = fopen(fileName, "r")))
+        if (0 != (status = rename(tempFileName, fileName)))
         {
-            fseek(file, 0, SEEK_END);
-            fileSize = ftell(file);
-            fseek(file, 0, SEEK_SET);
-
-            for (i = 0; i < fileSize; i++)
-            {
-                if (what == (next = fgetc(file)))
-                {
-                    numberOf += 1;
-                }
-                else if (EOF == next)
-                {
-                    break;
-                }
-            }
-
-            fclose(file);
+            OsConfigLogError(log, "ReplaceMarkedLinesInFile: rename('%s' to '%s') failed with %d", tempFileName, fileName, errno);
+            status = (0 == (status = errno)) ? ENOENT : errno;

```

Comment:
Could the inner `status = errno` be replaced with `errno`?

Reply:
Could. The intent was to avoid case when errno is reset. Will fi here and the other places.